src/Aspire.Hosting.Azure.Storage/AzureBlobStorageContainerHealthCheck.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Hosting;
+
+internal sealed class AzureBlobStorageContainerHealthCheck(BlobContainerClient blobContainerClient) : IHealthCheck
+{
+    private readonly BlobContainerClient _blobServiceClient = blobContainerClient ?? throw new ArgumentNullException(nameof(blobContainerClient));
+
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await _blobServiceClient.CreateIfNotExistsAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
> In general, I dislike using health checks to mutate state. They should just be used to check health, not make something "healthy".

Could the creation of the container be done outside, when the client is first created?

Reply:
Updated